lib/RawSource.js

```diff
@@ -16,9 +20,14 @@ class RawSource extends Source {
 			throw new TypeError("argument 'value' must be either string or Buffer");
 		}
 		this._valueIsBuffer = !convertToString && isBuffer;
-		this._value = convertToString && isBuffer ? undefined : value;
+		this._value =
+			convertToString && isBuffer
+				? undefined
+				: typeof value === "string"
+				? internString(value)
+				: value;
 		this._valueAsBuffer = isBuffer ? value : undefined;
-		this._valueAsString = isBuffer ? undefined : value;
+		this._valueAsString = isBuffer ? undefined : internString(value);
```

Comment:
This can avoid to call `internString` twice

Reply:
Done.